components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/CarbonMemoryAgentService.java

```diff
@@ -0,0 +1,164 @@
+    }
+
+    /**
+     * Class to log memory utilization periodically.
+     */
+    public static class Reporter extends TimerTask {
+
+        enum Attribute {
+            heap_memory_used_mb("Heap Memory Used"),
+            heap_memory_committed_mb("Heap Memory Committed"),
+            heap_memory_max_mb("Heap Memory Max"),
+            nonheap_memory_used_mb("Non-Heap Memory Used"),
+            nonheap_memory_committed_mb("Non-Heap Memory Committed"),
+            nonheap_memory_max_mb("Non-Heap Memory Maximum"),
+            total_threads("Total Threads"),
+            daemon_threads("Daemon Threads"),
+            nondaemon_threads("Non-Daemon Threads"),
+            internal_threads("Internal Threads (GC, etc)"),
+            mapped_buffers_count("Mapped Buffers Count");
+
+            public final String name;
+
+            Attribute(String name) {
+                this.name = name;
+            }
+        }
+
+        @Override
+        public void run() {
+
+            EnumMap<Attribute, Long> attributes = new EnumMap<Attribute, Long>(Attribute.class);
+            getMemoryUtilization(attributes);
+            getThreadUtilization(attributes);
+            printUserReport(attributes);
+        }
+
+        private void printUserReport(EnumMap<Attribute, Long> attributes) {
+
+            formatAndOutput("JVM Memory Usage     (Heap): used: %dM committed: %dM max:%dM",
+                    attributes.get(Attribute.heap_memory_used_mb), attributes.get(Attribute.heap_memory_committed_mb),
+                    attributes.get(Attribute.heap_memory_max_mb));
+            formatAndOutput("JVM Memory Usage (Non-Heap): used: %dM committed: %dM max:%dM",
+                    attributes.get(Attribute.nonheap_memory_used_mb), attributes.get(Attribute.nonheap_memory_committed_mb),
+                    attributes.get(Attribute.nonheap_memory_max_mb));
+            formatAndOutput("JVM Threads                : total: %d daemon: %d non-daemon: %d internal: %d",
+                    attributes.get(Attribute.total_threads), attributes.get(Attribute.daemon_threads),
+                    attributes.get(Attribute.nondaemon_threads), attributes.get(Attribute.internal_threads));
+        }
+
+        private void getThreadUtilization(EnumMap<Attribute, Long> attributes) {
+
+            int internal = hotspotThreadMBean.getInternalThreadCount();
+            int totalThreads = threadMXBean.getThreadCount();
+            int daemonThreads = threadMXBean.getDaemonThreadCount();
+            int nonDaemon = totalThreads - daemonThreads;
+            attributes.put(Attribute.total_threads, new Long(totalThreads + internal));
+            attributes.put(Attribute.daemon_threads, new Long(daemonThreads));
+            attributes.put(Attribute.nondaemon_threads, new Long(nonDaemon));
+            attributes.put(Attribute.internal_threads, new Long(internal));
+        }
+
+        private void getMemoryUtilization(EnumMap<Attribute, Long> attributes) {
```

Comment:
Instead of modifying the input, I prefer if u can return a new map so that u can aggregate when u need as per your requirements.

Reply:
fixed and made the method public to provide the capability to use else where